interface/external/pkmn_external.go

```diff
@@ -0,0 +1,163 @@
+package external
+
+import (
+	"encoding/csv"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+
+	"github.com/jvillarreal-w/academy-go-q42021/domain/external"
+	"github.com/jvillarreal-w/academy-go-q42021/domain/model"
+	"github.com/jvillarreal-w/academy-go-q42021/interface/context"
+	u "github.com/jvillarreal-w/academy-go-q42021/utils"
+)
+
+const url = "https://pokeapi.co/api/v2/pokemon"
+
+type pokemonExternal struct {
+}
+
+type PokemonExternal interface {
+	GetExternalPokemon(p []*model.Pokemon, c context.Context) ([]*model.Pokemon, error)
+	SaveExternalPokemon([]*model.Pokemon)
+}
+
+func NewPokemonExternal() PokemonExternal {
+	return &pokemonExternal{}
+}
+
+func (pe *pokemonExternal) GetExternalPokemon(p []*model.Pokemon, c context.Context) ([]*model.Pokemon, error) {
+	request, err := http.Get(fmt.Sprintf("%v?limit=50", url))
```

Comment:
I would suggest reading this article to see how you can refactor a bit to be able to mock http requests. We want to unit test this function and we don't actually want to make http requests when doing so:

https://www.thegreatcodeadventure.com/mocking-http-requests-in-golang/